pkg/server/context.go

```diff
@@ -38,6 +45,16 @@ func NewContext(ctx context.Context, cfg string) (*Context, error) {
 		return nil, err
 	}
 
+	// Construct a custom user-agent string for the apply client used by the deploy controller
+	// so that we can track which node's deploy controller most recently modified a resource.
+	nodeName := os.Getenv("NODE_NAME")
```

Comment:
is there a possibility that this can be empty since NODE_NAME is populated in the agent path code?

Reply:
Its already set by the time we get here, since the controller block until after startup is complete